src/ContractInteract/EIP20Gateway.js

```diff
@@ -0,0 +1,645 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+'use strict';
+
+const Web3 = require('web3');
+const BN = require('bn.js');
+const Contracts = require('../Contracts');
+const Utils = require('../../src/utils/Utils');
+const EIP20Token = require('../../src/ContractInteract/EIP20Token');
+const Anchor = require('../../src/ContractInteract/Anchor');
+
+/**
+ * Contract interact for EIP20Gateway.
+ */
+class EIP20Gateway {
+  /**
+   * Constructor for EIP20Gateway.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} gatewayAddress Gateway contract address.
+   */
+  constructor(web3, gatewayAddress) {
+    if (web3 instanceof Web3) {
+      this.web3 = web3;
+    } else {
+      const err = new TypeError(
+        "Mandatory Parameter 'web3' is missing or invalid",
+      );
+      throw err;
+    }
+
+    if (!Web3.utils.isAddress(gatewayAddress)) {
+      const err = new TypeError(
+        "Mandatory Parameter 'gatewayAddress' is missing or invalid.",
+      );
+      throw err;
+    }
+
+    this.gatewayAddress = gatewayAddress;
+
+    this.contract = Contracts.getEIP20Gateway(this.web3, this.gatewayAddress);
+
+    if (!this.contract) {
+      const err = new TypeError(
+        `Could not load Gateway contract for: ${this.gatewayAddress}`,
+      );
+      throw err;
+    }
+
+    this.proveGateway = this.proveGateway.bind(this);
+    this._proveGatewayRawTx = this._proveGatewayRawTx.bind(this);
+    this.stake = this.stake.bind(this);
+    this._stakeRawTx = this._stakeRawTx.bind(this);
+    this.progressStake = this.progressStake.bind(this);
+    this._progressStakeRawTx = this._progressStakeRawTx.bind(this);
+    this.getBounty = this.getBounty.bind(this);
+    this.getBaseToken = this.getBaseToken.bind(this);
+    this.getValueToken = this.getValueToken.bind(this);
+    this.getNonce = this.getNonce.bind(this);
+    this.getStateRootProviderAddress = this.getStateRootProviderAddress.bind(
+      this,
+    );
+    this.getInboxMessageStatus = this.getInboxMessageStatus.bind(this);
+    this.getOutboxMessageStatus = this.getOutboxMessageStatus.bind(this);
```

Comment:
Methods are missing, e.g. `getAnchor()`. Please check that all classes have all methods bound to them.